core/src/context.rs

```diff
@@ -281,78 +367,178 @@ impl Context {
+
+    fn collect_module_arguments(
+        &self,
+        args: &ModuleArguments,
+        module_name: &str,
+        output_format: &OutputFormat,
+    ) -> Result<HashMap<String, String>, CoreError> {
+        let empty_vec = vec![];
+        let mut pos_args = args.positioned.as_ref().unwrap_or(&empty_vec).iter();
+        let mut named_args = args.named.clone().unwrap_or_default();
+        let mut collected_args = HashMap::new();
+
+        // Get info about what args this parent node supports
+        let empty_vec = vec![];
+        let args_info = self
+            .get_transform_info(module_name, output_format)
+            .map_or(&empty_vec, |info| info.arguments.as_ref());
+
+        for arg_info in args_info {
+            let ArgInfo {
+                name,
+                default,
+                description: _,
+            } = arg_info;
+
+            // First empty the positional arguments
+            if let Some(value) = pos_args.next() {
+                // Check that this key is not repeated later too
+                if named_args.contains_key(name) {
+                    return Err(CoreError::RepeatedArgument(
+                        name.to_string(),
                         module_name.to_string(),
                     ));
                 }
+                collected_args.insert(name.to_string(), value.clone());
+                continue;
+            }
 
-                Ok(JsonEntry::Module {
-                    name: module_name.clone(),
-                    arguments: collected_args,
-                    data: body.clone(),
-                    inline: *one_line,
-                })
+            // Check if it was specified as a named key=value pair
+            if let Some(value) = named_args.remove(name) {
+                collected_args.insert(name.to_string(), value.clone());
+                continue;
+            }
+
+            // Use the default value as a fallback
+            if let Some(value) = default {
+                collected_args.insert(name.to_string(), value.clone());
+                continue;
             }
+
+            // Oops, the user seem to be missing a required argument
+            return Err(CoreError::MissingArgument(
+                name.to_string(),
+                module_name.to_string(),
+            ));
+        }
+
+        if let Some(value) = pos_args.next() {
+            return Err(CoreError::InvalidArgument(
+                value.to_string(),
+                module_name.to_string(),
+            ));
+        }
+
+        if let Some((key, _)) = named_args.iter().next() {
+            return Err(CoreError::InvalidArgument(
+                key.clone(),
+                module_name.to_string(),
+            ));
+        }
+        Ok(collected_args)
+    }
+}
+
+/// An enum containing one of two data types. Note that this is similar to `Result`, minus the
+/// convention that the right value is an error type.
+#[derive(Debug, Clone)]
+pub enum Either<L, R> {
```

Comment:
Perhaps we could have a special purpose type like `TransformOutput` instead since that's our only use case of `Either`. But I also like having a general either, so I'm not super convinced about that idea.  But, you might want to consider using the [either crate](https://crates.io/crates/either) instead.. Then we would have a bit less code to maintain.

Reply:
I switched to use the Either crate. I thought it was the perfect fit for this use case, and that's why I opted for that instead of a custom type.